src/components/structures/RoomView.js

```diff
@@ -159,6 +159,7 @@ module.exports = React.createClass({
         MatrixClientPeg.get().on("Room.name", this.onRoomName);
         MatrixClientPeg.get().on("Room.accountData", this.onRoomAccountData);
         MatrixClientPeg.get().on("RoomState.members", this.onRoomStateMember);
+        MatrixClientPeg.get().on("RoomMember.membership", this.onRoomMemberMembership);
```

Comment:
Could do, this quite nicely separates out stuff related to our room membership into a separate function though. Don't really mind either way.